internal/controller/api/hiveextension/v1beta1/agentclusterinstall_types.go

```diff
@@ -0,0 +1,180 @@
+	// +optional
+	WorkerAgentsReady int `json:"workerAgentsReady,omitempty"`
+
+	ConnectivityMajorityGroups string `json:"connectivityMajorityGroups,omitempty"`
+}
+
+// Networking defines the pod network provider in the cluster.
+type Networking struct {
+	// MachineNetwork is the list of IP address pools for machines.
+	// +optional
+	MachineNetwork []MachineNetworkEntry `json:"machineNetwork,omitempty"`
+
+	// ClusterNetwork is the list of IP address pools for pods.
+	// Default is 10.128.0.0/14 and a host prefix of /23.
+	//
+	// +optional
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork,omitempty"`
+
+	// ServiceNetwork is the list of IP address pools for services.
+	// Default is 172.30.0.0/16.
+	// NOTE: currently only one entry is supported.
+	//
+	// +kubebuilder:validation:MaxItems=1
+	// +optional
+	ServiceNetwork []string `json:"serviceNetwork,omitempty"`
+}
+
+// MachineNetworkEntry is a single IP address block for node IP blocks.
+type MachineNetworkEntry struct {
+	// CIDR is the IP block address pool for machines within the cluster.
+	CIDR string `json:"cidr"`
+}
+
+// ClusterNetworkEntry is a single IP address block for pod IP blocks. IP blocks
+// are allocated with size 2^HostSubnetLength.
+type ClusterNetworkEntry struct {
+	// CIDR is the IP block address pool.
+	CIDR string `json:"cidr"`
+
+	// HostPrefix is the prefix size to allocate to each node from the CIDR.
+	// For example, 24 would allocate 2^8=256 adresses to each node. If this
+	// field is not used by the plugin, it can be left unset.
+	// +optional
+	HostPrefix int32 `json:"hostPrefix,omitempty"`
+}
+
+// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+type ProvisionRequirements struct {
+
+	// ControlPlaneAgents is the number of matching approved and ready Agents with the control plane role
+	// required to launch the install. Must be either 1 or 3.
+	ControlPlaneAgents int `json:"controlPlaneAgents"`
+
+	// WorkerAgents is the minimum number of matching approved and ready Agents with the worker role
+	// required to launch the install.
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	WorkerAgents int `json:"workerAgents,omitempty"`
+}
+
+// HyperthreadingMode is the mode of hyperthreading for a machine.
+// +kubebuilder:validation:Enum="";Enabled;Disabled
+type HyperthreadingMode string
+
+const (
+	// HyperthreadingEnabled indicates that hyperthreading is enabled.
+	HyperthreadingEnabled HyperthreadingMode = "Enabled"
+	// HyperthreadingDisabled indicates that hyperthreading is disabled.
+	HyperthreadingDisabled HyperthreadingMode = "Disabled"
+)
+
+// AgentMachinePool is a pool of machines to be installed.
+type AgentMachinePool struct {
+	// Hyperthreading determines the mode of hyperthreading that machines in the
+	// pool will utilize.
+	// Default is for hyperthreading to be enabled.
+	//
+	// +optional
+	Hyperthreading HyperthreadingMode `json:"hyperthreading,omitempty"`
```

Comment:
@filanov regarding adding a Name here, in the installer world it's used for the naming of MachineSets in the cluster, infra in the cloud, etc. I'm not sure how these concepts relate to Assisted. Perhaps you may want to change it from being an array if you just want to blanket disable hyperthreading for all workers.

Reply:
MGMT-6022 will attend AgentMachinePool